tortoise/models.py

```diff
@@ -724,6 +741,47 @@ def update_from_dict(self, data: dict) -> MODEL:
         self._set_kwargs(data)
         return self  # type: ignore
 
+    @classmethod
+    def register_listener(cls, signal: Signals, listener: Callable):
+        assert callable(listener), "listener must be callable!"
+        cls._listeners.setdefault(signal, {})
+        cls._listeners.get(signal).setdefault(cls, [])
+        if listener not in cls._listeners.get(signal).get(cls):
+            cls._listeners.get(signal).get(cls).append(listener)
+
+    async def _pre_delete(self, using_db: Optional[BaseDBAsyncClient] = None,) -> None:
+        listeners = []
+        for listener in self._listeners.get(Signals.pre_delete, {}).get(self.__class__, []):
+            listeners.append(listener(self.__class__, self, using_db,))
+        await asyncio.gather(*listeners)
+
+    async def _post_delete(self, using_db: Optional[BaseDBAsyncClient] = None,) -> None:
+        listeners = []
+        for listener in self._listeners.get(Signals.post_delete, {}).get(self.__class__, []):
+            listeners.append(listener(self.__class__, self, using_db,))
+        await asyncio.gather(*listeners)
+
+    async def _pre_save(
+        self,
+        using_db: Optional[BaseDBAsyncClient] = None,
+        update_fields: Optional[List[str]] = None,
+    ) -> None:
+        listeners = []
+        for listener in self._listeners.get(Signals.pre_save, {}).get(self.__class__, []):
+            listeners.append(listener(self.__class__, self, using_db, update_fields))
```

Comment:
Why do you send the Class and the Instance for each callback?

Reply:
multiple Class can register same signal like ``pre_save(*senders)``,and in listener,we can know which Class send that signal,and get the Class instance,maybe we can know Class with ``instance.__class__`` and without Class pass in?

Reply:
Also, all the class methods still exist on the instance.
One can do <instance x>.filter(...)
Although it doesn't read right if you do that.